framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -78,95 +79,98 @@ import { customComponents } from './custom-components'
  * ?)
  */
 
-export const CreatableSelectDropdown = <T extends string = string> ({
+export const CreatableSelectDropdown = <T extends string = string>({
   standardOptions,
   initialPlaceholder = 'Select or create...',
-  addOptionPlaceholder = 'Enter text...',
+  addOptionPlaceholder = 'Select or create...',
   creationOption = {
-    label: 'Add option...',
-    value: 'Add option...',
+    label: 'Add field',
+    value: 'Add field',
     isCreation: true,
   },
   onOptionChange,
-  width = '100%',
   variant = 'outline',
   defaultValue,
   value,
   menuPlacement = 'bottom',
 }: CreatableSelectDropdownProps<T>) => {
-  const initialSelectedOption = useMemo(() => {
-    if (!isNil(value)) {
-      return standardOptions.find((option) => option.value === value) ?? null
-    } if (!isNil(defaultValue)) {
-      return standardOptions.find((option) => option.value === defaultValue) ?? null
-    }
-    return null
-  }, [ value ]
-  )
+  const [ selectedOption, setSelectedOption ] = useState<Option | null>(() => {
+    const targetValue = value ?? defaultValue ?? null
+    return targetValue
+      ? standardOptions.find((option) => option.value === targetValue) ?? null
+      : null
+  })
 
-  const [ selectedOption, setSelectedOption ] =
-    useState<Option | null>(initialSelectedOption)
-  const [ newOptionText, setNewOptionText ] = useState('')
-  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] =
+    useState(initialPlaceholder)
   const [ createdOptions, setCreatedOptions ] = useState<Option[]>([])
+  const [ createNewOption, setCreateNewOption ] = useState<boolean>(false)
+  const [ addFieldInputValue, setAddFieldInputValue ] = useState<string>('')
+
+  const toggleCreateNewOption = () => {
+    setCreateNewOption(not)
+  }
```

Comment:
Let's not toggle. Toggling here is quite dangerous and may break if a small change is introduced. Manually setting state value to true/false will give you more control.